agent/engine/docker_task_engine.go

```diff
@@ -602,41 +611,119 @@ func (engine *DockerTaskEngine) startContainer(task *api.Task, container *api.Co
 	containerMap, ok := engine.state.ContainerMapByArn(task.Arn)
 	if !ok {
 		return DockerContainerMetadata{
-			Error: CannotStartContainerError{fmt.Errorf("Container belongs to unrecognized task %s", task.Arn)},
+			Error: CannotStartContainerError{errors.Errorf("Container belongs to unrecognized task %s", task.Arn)},
 		}
 	}
 
 	dockerContainer, ok := containerMap[container.Name]
 	if !ok {
 		return DockerContainerMetadata{
-			Error: CannotStartContainerError{fmt.Errorf("Container not recorded as created")},
+			Error: CannotStartContainerError{errors.Errorf("Container not recorded as created")},
 		}
 	}
 	return client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
 }
 
 func (engine *DockerTaskEngine) provisionContainerResources(task *api.Task, container *api.Container) DockerContainerMetadata {
-	seelog.Infof("Task [%v]: Setting up container resources for container [%v]", task.String(), container.String())
+	seelog.Infof("Task [%s]: Setting up container resources for container [%s]", task.String(), container.String())
+
+	cniConfig, err := engine.BuildCNIConfigFromTaskContainer(task, container)
+	if err != nil {
+		return DockerContainerMetadata{
+			DockerID: cniConfig.ContainerID,
+			Error:    NetworkNamespaceError{errors.Wrap(err, "provisionContainerResources: build cni configuration error")},
+		}
+	}
+	// Invoke the libcni to config the network namespace for the container
+	err = engine.cniClient.SetupNS(cniConfig)
+	if err != nil {
+		log.Error("engine: Set up pause container namespace failed, err: %v, task: %s", err, task.String())
+		return DockerContainerMetadata{
+			DockerID: cniConfig.ContainerID,
+			Error:    NetworkNamespaceError{errors.Wrap(err, "provisionContainerResources: setup network namespace error")},
+		}
+	}
+
+	return DockerContainerMetadata{
+		DockerID: cniConfig.ContainerID,
+	}
+}
+
+// cleanupPauseContainer will clean up the network namespace of pause container
+func (engine *DockerTaskEngine) cleanupPauseContainer(task *api.Task, container *api.Container) error {
```

Comment:
It doesn't look like this is cleaning up the pause container, just cleaning up the namespace.  Maybe this should be `cleanupPauseContainerResources`?